core/modules/qana/QservRestrictorPlugin.cc

```diff
@@ -571,14 +574,36 @@ QservRestrictorPlugin::_newSecIndexRestrictor(
         return query::QsRestrictor::Ptr();
     }
 
+    // Get the director table (or an empty string)
+    std::string dirTable = context.cssFacade->getDirTable(cr->db, cr->table);
+
     // Build the QsRestrictor
     query::QsRestrictor::Ptr p = std::make_shared<query::QsRestrictor>();
     p->_name = "sIndex";
     // sIndex has paramers as follows:
     // db, table, column, val1, val2, ...
-    p->_params.push_back(cr->db);
-    p->_params.push_back(cr->table);
-    p->_params.push_back(cr->column);
+    if (!dirTable.empty()) {
+        std::string dirDb = context.cssFacade->getDirDb(cr->db, cr->table);
+        if (dirDb.empty()) {
+            dirDb = cr->db;
+        }
+        std::string dirCol = context.cssFacade->getDirColName(cr->db, cr->table);
+        if (dirCol.empty()) {
+            LOGF_ERROR("dirTable set, dirCol missing for %1%.%2%" % cr->db % cr->table);
+            return query::QsRestrictor::Ptr();
+        }
+        LOGF_DEBUG("Using dirDb %1%, dirTable %2%, dirCol %3% as sIndex for %4%.%5%." %
+                   dirDb % dirTable % dirCol % cr->db % cr->table);
+        p->_params.push_back(dirDb);
+        p->_params.push_back(dirTable);
+        p->_params.push_back(dirCol);
+    } else {
+        LOGF_DEBUG("Using %1%, %2%, %3% as sIndex." %
+                   cr->db % cr->table % cr->column);
+        p->_params.push_back(cr->db);
+        p->_params.push_back(cr->table);
+        p->_params.push_back(cr->column);
+    }
```

Comment:
Forget the || vs or, it is a typo, (I was just tweaking python code).

The way I imagined it was:
    dirDb = getDirDb("myDb", "Source");           // say it is "", so we default to current
    dirTable = getDirTable("myDb", "Source"); // dirTable gets set to "Object" 
    dirCol = getDirCol("myDb", "Source");        // dirCol gets set to "objectId"
and there we have everything we need to know about dirDb/dirTable/dirCol

I think I hear you are saying that after
    dirTable = getDirTable("myDb", "Source"); // dirTable gets set to "Object" 
if the dirTable has been set to something different than current table, the
    dirCol = getDirCol("myDb", "Source");        // dirCol gets set to "objectId"
should really become
    dirCol = getDirCol("myDb", "Object");        // dirCol gets set to "objectId"
Hmm... I guess we could chain it like that, but somehow I always imagined that we should set dirTable AND dirCol together, and we don't need to follow the chain: ("what is my dirTable?", "ok, it is X", "let me go to X and check its dirCol"... ). If we do that, then I'd argue that if we set dirTable, we should DISALLOW dirCol, and it should be required to go to dirTable to get dirCol.